src/components/homepage/calculator-preview.tsx

```diff
@@ -0,0 +1,367 @@
+  CALCULATOR_CONSTANTS.DEDUCTIONS.reduce(
+    (sum, deduction) => sum + deduction.employeeContribution,
+    0
+  ).toFixed(3)
+);
+const companyDeduction = Number(
+  CALCULATOR_CONSTANTS.DEDUCTIONS.reduce(
+    (sum, deduction) => sum + deduction.employerContribution,
+    0
+  ).toFixed(3)
+);
+const maxOneChildrenUnderDeduction =
+  CALCULATOR_CONSTANTS.MAX_ONE_CHILDREN_UNDER_DEDUCTION;
+const maxOneChildrenOverDeduction =
+  CALCULATOR_CONSTANTS.MAX_ONE_CHILDREN_OVER_DEDUCTION;
+
+//total children rates based on numbers from form
+function calculateChildAllowanceRate(totalChildren: number): number {
+  if (totalChildren > 6) {
+    return CALCULATOR_CONSTANTS.CHILD_ALLOWANCE_RATES.find(
+      (rate) => rate.numberOfChildren === 6
+    )!.rate;
+  }
+  const rate = CALCULATOR_CONSTANTS.CHILD_ALLOWANCE_RATES.find(
+    (rate) => rate.numberOfChildren === totalChildren
+  );
+  return rate ? rate.rate : 0;
+}
+
+// calculate NET WAGE
+function calculateNetWage(
+  grossWage: number,
+  nonTaxablePart: boolean,
+  childrenUnder18: number,
+  childrenOver18: number
+): {
+  netWage: number;
+  deductions: number;
+  taxableIncome: number;
+  nonTaxableAmount: number;
+  incomeWithoutDeductions: number;
+  incomeTaxAmount: number;
+  taxBonus: number;
+} {
+  const deductions = Number((grossWage * employeeDeduction).toFixed(2));
+  const nonTaxableAmount = nonTaxablePart ? nonTaxableAmountOfTaxBasis : 0;
+  const incomeWithoutDeductions = grossWage - deductions;
+  const taxableIncome = Number(
+    (grossWage - deductions - nonTaxableAmount).toFixed(2)
+  );
+  const incomeTaxAmount = Number((taxableIncome * incomeTax).toFixed(2));
+  const childrenTotal = childrenUnder18 + childrenOver18;
+  const taxBonusAmount = Number(
+    (
+      (grossWage - deductions) *
+      calculateChildAllowanceRate(childrenTotal)
+    ).toFixed(2)
+  );
+  const maxTaxBonus =
+    childrenOver18 * maxOneChildrenOverDeduction +
+    childrenUnder18 * maxOneChildrenUnderDeduction;
+  const taxBonus = taxBonusAmount > maxTaxBonus ? maxTaxBonus : taxBonusAmount;
+  const netWage = Number(
+    (grossWage - deductions - taxableIncome * incomeTax + taxBonus).toFixed(2)
+  );
+
+  return {
+    netWage: netWage,
+    deductions: deductions,
+    taxableIncome: taxableIncome,
+    nonTaxableAmount: nonTaxableAmount,
+    incomeWithoutDeductions: incomeWithoutDeductions,
+    incomeTaxAmount: incomeTaxAmount,
+    taxBonus: taxBonus,
+  };
+}
+
+//form schema
+const formSchema = z.object({
+  grossWage: z.number().min(minWage, {
+    message: "Gross wage must be at least minimum wage (750€)",
+  }),
+  nonTaxablePart: z.boolean().optional(),
+  childrenUnder18: z
+    .number()
+    .int()
+    .min(0, {
+      message: "Must be at least 0",
+    })
+    .max(10, { message: "Can't be more than 10" }),
+  childrenOver18: z
+    .number()
+    .int()
+    .min(0, {
+      message: "Must be at least 0",
+    })
+    .max(10, { message: "Can't be more than 10" }),
+});
```

Comment:
Form schema si vytiahni do extra súboru, prípadne si vytvor schemas folder a šupni to tam. 

Uplne ideálne je mať vždy component folder (v tomto prípade calculator-preview), v ňom mať component file (v tomto prípade calculator-preview.tsx) a veďla neho file schema.ts, v ktorom budeš mať túto schému (ideálne ju premenuj na calculatorPreviewSchema) a typ, ktorý si z nej inferneš.